File: ActiveSense/modules/active-sense-pose/ios/ActiveSensePoseModule.swift
import ExpoModulesCore

// ActiveSensePoseModule exposes the iOS camera pose view to React Native.
public class ActiveSensePoseModule: Module {
  public func definition() -> ModuleDefinition {
    // This name must match requireNativeView('ActiveSensePose') in TypeScript.
    Name("ActiveSensePose")

    View(ActiveSensePoseView.self) {
      // JS listens to landmarks for tracking and status for setup messages.
      Events("onLandmarks", "onStatus")

      // React controls whether the camera session should run.
      Prop("enabled") { (view: ActiveSensePoseView, enabled: Bool) in
        view.setEnabled(enabled)
      }

      // React can switch between front and back camera.
      Prop("cameraFacing") { (view: ActiveSensePoseView, cameraFacing: String) in
        view.setCameraFacing(cameraFacing)
      }

      // The preview can be restarted from JS if the camera pipeline needs a reset.
      AsyncFunction("restart") { (view: ActiveSensePoseView) in
        view.restart()
      }
    }
  }
}

File: ActiveSense/modules/active-sense-pose/android/src/main/java/com/activesense/pose/ActiveSensePoseModule.kt
package com.activesense.pose

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

// ActiveSensePoseModule exposes the Android camera pose view to React Native.
class ActiveSensePoseModule : Module() {
  override fun definition() = ModuleDefinition {
    // This name must match requireNativeView("ActiveSensePose") in TypeScript.
    Name("ActiveSensePose")

    View(ActiveSensePoseView::class) {
      // JS listens to pose frames and small lifecycle status messages.
      Events("onLandmarks", "onStatus")

      // React controls whether the native camera should run.
      Prop("enabled") { view: ActiveSensePoseView, enabled: Boolean ->
        view.setEnabled(enabled)
      }

      // React can switch between front and back lenses.
      Prop("cameraFacing") { view: ActiveSensePoseView, cameraFacing: String ->
        view.setCameraFacing(cameraFacing)
      }

      // Expose a reset hook for camera recovery.
      AsyncFunction("restart") { view: ActiveSensePoseView ->
        view.restart()
      }

      // Release CameraX and MediaPipe resources when React destroys the view.
      OnViewDestroys { view: ActiveSensePoseView ->
        view.cleanup()
      }
    }
  }
}
